src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1144,6 +1163,7 @@ private void EnforceDoesNotReturn(SyntaxNode? syntaxOpt)
                 NullableWalker.AnalyzeIfNeeded(
                     compilation,
                     method,
+                    initializers.BoundInitializers.SelectAsArray(initializer => initializer.Syntax),
```

Comment:
>initializers.BoundInitializers.SelectAsArray(initializer => initializer.Syntax) [](start = 20, length = 79)

It feels like the body of the constructor itself should be taken into consideration as well. I.e. if we have enabled analysis within the body, we should analyze initializers as well.

Reply:
Perhaps we should force analysis if any constructor that includes initializers needs the analysis.

---
In reply to: [544634546](https://github.com/dotnet/roslyn/pull/49953#discussion_r544634546) [](ancestors = 544634546)